sql/src/main/java/io/crate/metadata/sys/SysSegmentsTableInfo.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.metadata.sys;
+
+import io.crate.action.sql.SessionContext;
+import io.crate.analyze.WhereClause;
+import io.crate.expression.reference.sys.shard.ShardSegment;
+import io.crate.metadata.ColumnIdent;
+import io.crate.metadata.RelationName;
+import io.crate.metadata.Routing;
+import io.crate.metadata.RoutingProvider;
+import io.crate.metadata.RowGranularity;
+import io.crate.metadata.expressions.RowCollectExpressionFactory;
+import io.crate.metadata.table.ColumnRegistrar;
+import io.crate.metadata.table.StaticTableInfo;
+import io.crate.types.ObjectType;
+import org.elasticsearch.cluster.ClusterState;
+import org.elasticsearch.cluster.node.DiscoveryNode;
+
+import java.util.Map;
+import java.util.function.Supplier;
+
+import static io.crate.execution.engine.collect.NestableCollectExpression.forFunction;
+import static io.crate.types.DataTypes.BOOLEAN;
+import static io.crate.types.DataTypes.INTEGER;
+import static io.crate.types.DataTypes.STRING;
+import static io.crate.types.DataTypes.LONG;
+
+public class SysSegmentsTableInfo extends StaticTableInfo<ShardSegment> {
+
+    public static final RelationName IDENT = new RelationName(SysSchemaInfo.NAME, "segments");
+
+    public static Map<ColumnIdent, RowCollectExpressionFactory<ShardSegment>> expressions(Supplier<DiscoveryNode> localNode) {
+        return columnRegistrar(localNode).expressions();
+    }
+
+    private static ColumnRegistrar<ShardSegment> columnRegistrar(Supplier<DiscoveryNode> localNode) {
+        return new ColumnRegistrar<ShardSegment>(IDENT, RowGranularity.DOC)
+            .register("table_schema", STRING, () -> forFunction(r -> r.getIndexParts().getSchema()))
+            .register("table_name", STRING, () -> forFunction(r -> r.getIndexParts().getTable()))
+            .register("shard_id", INTEGER, () -> forFunction(ShardSegment::getShardId))
+            .register("_node", ObjectType.builder()
```

Comment:
In `sys.shards` and `sys.jobs` we've a `node` column instead of `_node`, I'd also go with `node` here for consistency.

Reply:
@seut insisted on using `_node` I went for `node` originally. If we want to go for consistency we should also use `schema_name` instead of `table_schema`.

Reply:
I'm fine with using `node` here as well. @mkleen to be fair, your original version contained `node_id` ;). Also `table_schema` is used everywhere else and is also the valid term related to the sql standard

Reply:
@seut `node_id` you are right, sorry.